codex/sales.nim

```diff
@@ -143,17 +102,15 @@ proc handleRequest(sales: Sales,
 proc load*(sales: Sales) {.async.} =
   let market = sales.context.market
 
-  # TODO: restore availability from disk
   let slotIds = await market.mySlots()
 
   for slotId in slotIds:
-    # TODO: this needs to be optimised
-    if request =? await market.getRequestFromSlotId(slotId):
-      let availability = sales.findAvailability(request.ask)
-      without slotIndex =? findSlotIndex(request.ask.slots,
-                                          request.id,
-                                          slotId):
-        raiseAssert "could not find slot index"
+    if (request, slotIndex) =? (await market.getActiveSlot(slotId)):
+      let availability = await sales.context.reservations.find(
```

Comment:
Why don't we look up the reservation based on the slot id? Couldn't this go wrong when there are multiple reservations with of the same size, duration and price?

Reply:
Whats the relationship between the slot and the reservation? Should there be one? The repostore currently simply counts used bytes vs reserved bytes, once a slot has been allocated and being stored, the "reserved" is simply counted as "used".

Reply:
An availability is marked as used by assigning the availability a slotId, which is stored in the repo:
https://github.com/status-im/nim-codex/blob/288c8cdcd881c122bd172beccef9f97525f59a6f/codex/sales/reservations.nim#L243-L248

A "used" availability is simply an availability that has a slotId assigned:
https://github.com/status-im/nim-codex/blob/288c8cdcd881c122bd172beccef9f97525f59a6f/codex/sales/reservations.nim#L103-L104